test/distributions/test_distributions.py

```diff
@@ -103,694 +99,701 @@ def is_all_nan(tensor):
-        },
-        {
-            'scale': torch.tensor([1.0]),
-            'alpha': -1.0
-        }
-    ]),
-    Example(Poisson, [
-        {
-            'rate': torch.tensor([-0.1], requires_grad=True),
-        },
-        {
-            'rate': -1.0,
-        }
-    ]),
-    Example(RelaxedBernoulli, [
-        {
-            'temperature': torch.tensor([1.5], requires_grad=True),
-            'probs': torch.tensor([1.7, 0.2, 0.4], requires_grad=True),
-        },
-        {
-            'temperature': torch.tensor([2.0]),
-            'probs': torch.tensor([-1.0]),
-        }
-    ]),
-    Example(RelaxedOneHotCategorical, [
-        {
-            'temperature': torch.tensor([0.5], requires_grad=True),
-            'probs': torch.tensor([[-0.1, 0.2, 0.7], [0.5, 0.3, 0.2]], requires_grad=True)
-        },
-        {
-            'temperature': torch.tensor([2.0]),
-            'probs': torch.tensor([[-1.0, 0.0], [-1.0, 1.1]])
-        }
-    ]),
-    Example(TransformedDistribution, [
-        {
-            'base_distribution': Normal(0, 1),
-            'transforms': lambda x: x,
-        },
-        {
-            'base_distribution': Normal(0, 1),
-            'transforms': [lambda x: x],
-        },
-    ]),
-    Example(Uniform, [
-        {
-            'low': torch.tensor([2.0], requires_grad=True),
-            'high': torch.tensor([2.0], requires_grad=True),
-        },
-        {
-            'low': torch.tensor([0.0], requires_grad=True),
-            'high': torch.tensor([0.0], requires_grad=True),
-        },
-        {
-            'low': torch.tensor([1.0], requires_grad=True),
-            'high': torch.tensor([0.0], requires_grad=True),
-        }
-    ]),
-    Example(Weibull, [
-        {
-            'scale': torch.tensor([0.0], requires_grad=True),
-            'concentration': torch.tensor([0.0], requires_grad=True)
-        },
-        {
-            'scale': torch.tensor([1.0], requires_grad=True),
-            'concentration': torch.tensor([-1.0], requires_grad=True)
-        }
-    ]),
-    Example(Wishart, [
-        {
-            'covariance_matrix': torch.tensor([[1.0, 0.0], [0.0, -2.0]], requires_grad=True),
-            'df': torch.tensor([1.5], requires_grad=True),
-        },
-        {
-            'covariance_matrix': torch.tensor([[1.0, 1.0], [1.0, -2.0]], requires_grad=True),
-            'df': torch.tensor([3.], requires_grad=True),
-        },
-        {
-            'covariance_matrix': torch.tensor([[1.0, 1.0], [1.0, -2.0]], requires_grad=True),
-            'df': 3.,
-        },
-    ]),
-    Example(ContinuousBernoulli, [
-        {'probs': torch.tensor([1.1, 0.2, 0.4], requires_grad=True)},
-        {'probs': torch.tensor([-0.5], requires_grad=True)},
-        {'probs': 1.00001},
-    ])
-]
+
+# Register all distributions for generic tests.
+def _get_examples(dtype=None):
+    if dtype is None:
+        dtype = torch.get_default_dtype()
+
+    with set_default_dtype(dtype):
```

Comment:
It should not be necessary to have this `dtype` arg if the caller is wrapping the call with a `set_default_dtype()` context manager. But I'm just trying this to see if for some reason this fixes the `test_cdf_icdf_inverse` failure in the [macos-12-py3-arm64 / test (default, 1, 3, macos-m1-12)](https://github.com/pytorch/pytorch/actions/runs/5884003265/job/15958041824?pr=107246#logs) job.

Reply:
It didn't work. Removing this.